### PR TITLE
Fix/spam message server zappy gui

### PR DESCRIPTION
**Pull request by mathieurobert1**
Fix the spam of Mct command by call it all the 20 / ticks second.

I also update tile when take and drop item.

More details : #174

### CODE REVIEW

File: gui/include/GameDatas/GameData.hpp [hunk 1]
```diff
@@ -239,14 +239,43 @@ class Gui::GameData {
...
+         * @param nb Number of bct command received.
+         */
+        void setnbBCTCommandReceived(std::size_t nb);
+
+        /**
+         * @brief Get the number of bct command received.
+         *
+         * @return std::size_t - Number of bct command received.
+         */
+        std::size_t getnbBCTCommandReceived();
```
Comment (Thomaltarix) [inline gui/include/GameDatas/GameData.hpp:254]: Maybe `set/getNbBTCCommandReceived`

File: gui/src/GameDatas/Inventory.cpp [hunk 2]
```diff
@@ -96,3 +96,25 @@ Gui::Inventory::Ressources &Gui::Inventory::getRessources(void)
...
+
+void Gui::Inventory::addResource(std::size_t resource, std::size_t quantity)
+{
+    if (resource == 0) {
+        _food += quantity;
+        return;
+    }
+    if (resource >= 1 && resource <= 6) {
+        _ressources[resource - 1] += quantity;
+    }
```
Comment (AxelF44) [inline gui/src/GameDatas/Inventory.cpp:108]:
```suggestion
    if (resource >= 1 && resource <= 6)
        _ressources[resource - 1] += quantity;
```

File: gui/src/GameDatas/Inventory.cpp [hunk 3]
```diff
@@ -96,3 +96,25 @@ Gui::Inventory::Ressources &Gui::Inventory::getRessources(void)
...
+
+void Gui::Inventory::removeResource(std::size_t resource, std::size_t quantity)
+{
+    if (resource == 0) {
+        _food = (_food < quantity) ? 0 : _food - quantity;
+        return;
+    }
+    if (resource >= 1 && resource <= 6) {
+        _ressources[resource - 1] = (_ressources[resource - 1] < quantity) ? 0 : _ressources[resource - 1] - quantity;
+    }
```
Comment (AxelF44) [inline gui/src/GameDatas/Inventory.cpp:119]:
```suggestion
    if (resource >= 1 && resource <= 6)
        _ressources[resource - 1] = (_ressources[resource - 1] < quantity) ? 0 : _ressources[resource - 1] - quantity;
```

File: gui/src/Engine/Engine.cpp [hunk 4]
```diff
@@ -71,3 +72,24 @@ void Gui::Engine::sendMessageUpdate()
...
+
+void Gui::Engine::updateMap()
+{
+    clock_t currentTick = clock();
+    std::pair<std::size_t, std::size_t> mapSize = _gameData.get()->getMapSize();
+
+    if (!(mapSize.first == 0 && mapSize.second == 0) && _gameData.get()->getnbBCTCommandReceived() >= mapSize.first * mapSize.second) {
+        if (20 / _gameData->getServerTick() >= 2 && (float)(currentTick - _gameData->getLastTickMctCommand()) / CLOCKS_PER_SEC > 20 / _gameData->getServerTick()) {
+            sendUpdateMapMessage();
+        } else if (20 / _gameData->getServerTick() < 2 && (float)(currentTick - _gameData->getLastTickMctCommand()) / CLOCKS_PER_SEC > 2) {
```
Comment (Thomaltarix) [inline gui/src/Engine/Engine.cpp:84]: The 20 should be a MACRO imo

File: gui/src/GameDatas/GameData.cpp [hunk 5]
```diff
@@ -205,3 +207,23 @@ void Gui::GameData::removeServerEgg(size_t id)
         }
     }
 }
+
+std::size_t Gui::GameData::getnbBCTCommandReceived()
```
Comment (AxelF44) [inline gui/src/GameDatas/GameData.cpp:211]:
```suggestion
std::size_t Gui::GameData::getnbBCTCommandReceived() const
```

**Review by AxelF44 — CHANGES_REQUESTED** [review]

File: gui/src/GameDatas/Inventory.cpp (hunk 2 above)
Comment (Thomaltarix) [inline gui/src/GameDatas/Inventory.cpp:108]: Useless brackets

File: gui/src/GameDatas/Inventory.cpp (hunk 3 above)
Comment (Thomaltarix) [inline gui/src/GameDatas/Inventory.cpp:119]: Same here

**Review by Thomaltarix — CHANGES_REQUESTED** [review]
Some changes to do

**Review by AxelF44 — APPROVED** [review]

**Review by Thomaltarix — APPROVED** [review]

**Pull request merged**